api/envoy/config/core/v3/address.proto

```diff
@@ -97,6 +97,11 @@ message SocketAddress {
   // allow both IPv4 and IPv6 connections, with peer IPv4 addresses mapped into
   // IPv6 space as ``::FFFF:<IPv4-address>``.
   bool ipv4_compat = 6;
+
+  // The Linux network namespace to bind the socket to. If this is set, Envoy will
+  // create the socket in the specified network namespace. Only supported on Linux.
```

Comment:
Is there any format for the network namespace file path we should follow, e.g., `/proc/<pid>/ns/net`? Maybe add more comments?

Reply:
Please also take into account named network namespaces, which are solved via bind mounts to `/var/run/netns/<name>`

Reply:
I'll have to follow this PR up with the implementation, so I'll beef up the comments with examples in the next patch.